cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -14,79 +14,141 @@
+};
+
+std::unique_ptr<tfrt::HostContext> OneHotEncoderTest::host_context = nullptr;
+
+std::unique_ptr<MLIRContext> OneHotEncoderTest::mlir_context = nullptr;
+
+TEST_F(OneHotEncoderTest, Param) {
+  RCReference<OneHotEncoderModel> model = tfrt::TakeRef(
+      host_context->Construct<OneHotEncoderModel>(host_context.get()));
+  model->setDropLast(false);
+  EXPECT_FALSE(model->getDropLast());
+  model->setDropLast(true);
+  EXPECT_TRUE(model->getDropLast());
 }
 
-TEST(OneHotEncoderTest, Transform) {
-  clink::OneHotEncoderModelDataProto model_data;
+TEST_F(OneHotEncoderTest, Transform) {
+  OneHotEncoderModelDataProto model_data;
   model_data.add_featuresizes(2);
   model_data.add_featuresizes(3);
   std::string model_data_str;
   model_data.SerializeToString(&model_data_str);
 
-  clink::OneHotEncoderModel model;
-  model.setDropLast(false);
-  llvm::Error err = model.setModelData(model_data_str);
-  EXPECT_EQ(!err, true);
+  RCReference<OneHotEncoderModel> model = tfrt::TakeRef(
+      host_context->Construct<OneHotEncoderModel>(host_context.get()));
+  model->setDropLast(false);
+  llvm::Error err = model->setModelData(std::move(model_data_str));
+  EXPECT_FALSE(err);
 
-  auto vector = model.transform(1, 0);
-  EXPECT_EQ(!vector.takeError(), true);
+  auto vector = model->transform(1, 0);
+  EXPECT_FALSE(vector.IsError());
   EXPECT_EQ(vector->get(1).get(), 1.0);
 
-  auto invalid_value_vector = model.transform(1, 5);
-  EXPECT_EQ(!invalid_value_vector.takeError(), false);
+  auto invalid_value_vector = model->transform(1, 5);
+  EXPECT_TRUE(invalid_value_vector.IsError());
 
-  auto invalid_index_vector = model.transform(5, 0);
-  EXPECT_EQ(!invalid_index_vector.takeError(), false);
+  auto invalid_index_vector = model->transform(5, 0);
+  EXPECT_TRUE(invalid_index_vector.IsError());
 }
 
-TEST(OneHotEncoderTest, Load) {
-  std::string dir_name = clink::test::createTemporaryFolder();
+TEST_F(OneHotEncoderTest, Load) {
+  test::TemporaryFolder tmp_folder;
 
   nlohmann::json params;
   params["paramMap"]["dropLast"] = "false";
 
-  std::ofstream params_output(dir_name + "/metadata");
-  params_output << params;
-  params_output.close();
-
-  clink::OneHotEncoderModelDataProto model_data;
+  OneHotEncoderModelDataProto model_data;
   model_data.add_featuresizes(2);
   model_data.add_featuresizes(3);
 
-  mkdir((dir_name + "/data").c_str(), S_IRWXU);
-  std::ofstream model_data_output(dir_name + "/data/" +
-                                  clink::test::generateRandomString());
-  model_data.SerializeToOstream(&model_data_output);
-  model_data_output.close();
+  test::saveMetaDataModelData(tmp_folder.getAbsolutePath(), params, model_data);
 
-  auto model = clink::OneHotEncoderModel::load(dir_name);
-  EXPECT_EQ(!model.takeError(), true);
+  auto model = OneHotEncoderModel::load(tmp_folder.getAbsolutePath(),
+                                        host_context.get());
+  EXPECT_TRUE((bool)model);
```

Comment:
`llvm::Expected` overrides the `bool` operator, so that it returns false if there is an error.

I'll change this to `EXPECT_FALSE((bool)model.takeError())`. Hope this can improve the code's readability.

Reply:
Sounds good. Could you add empty space after `(bool)`?

Reply:
Like discussed offline, I tried adding empty spaces here and found that clang-format would remove the whitespace. So let's keep it without whitespaces as it is.

Reply:
Sounds good.